lib/IMAP/MailboxSync.php

```diff
@@ -0,0 +1,141 @@
+<?php declare(strict_types=1);
+
+/**
+ * @copyright 2019 Christoph Wurst <christoph@winzerhof-wurst.at>
+ *
+ * @author 2019 Christoph Wurst <christoph@winzerhof-wurst.at>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Mail\IMAP;
+
+use OCA\Mail\Account;
+use OCA\Mail\Db\MailAccountMapper;
+use OCA\Mail\Db\Mailbox;
+use OCA\Mail\Db\MailboxMapper;
+use OCA\Mail\Folder;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\ILogger;
+
+class MailboxSync {
+
+	/** @var MailboxMapper */
+	private $mailboxMapper;
+
+	/** @var FolderMapper */
+	private $folderMapper;
+
+	/** @var MailAccountMapper */
+	private $mailAccountMapper;
+
+	/** @var IMAPClientFactory */
+	private $imapClientFactory;
+
+	/** @var ITimeFactory */
+	private $timeFactory;
+
+	/** @var ILogger */
+	private $logger;
+
+	public function __construct(MailboxMapper $mailboxMapper,
+								FolderMapper $folderMapper,
+								MailAccountMapper $mailAccountMapper,
+								IMAPClientFactory $imapClientFactory,
+								ITimeFactory $timeFactory,
+								ILogger $logger) {
+		$this->mailboxMapper = $mailboxMapper;
+		$this->folderMapper = $folderMapper;
+		$this->mailAccountMapper = $mailAccountMapper;
+		$this->imapClientFactory = $imapClientFactory;
+		$this->timeFactory = $timeFactory;
+		$this->logger = $logger;
+	}
+
+	public function sync(Account $account): void {
+		if ($account->getMailAccount()->getLastMailboxSync() >= ($this->timeFactory->getTime() - 7200)) {
```

Comment:
I assume it is future work to be able to force a sync? Because otherwise I can already see the bug reports come in :wink:

Reply:
Yep, I could imagine something like an entry in the account's dropdown with "refresh folder list". But for 99.9% of all cases it's relatively rare that folders change, I guess